src/components/FullscreenLoadingIndicator.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import {ActivityIndicator, StyleSheet, View} from 'react-native';
+import styles from '../styles/styles';
+import themeColors from '../styles/themes/default';
+
+type FullScreenLoadingIndicatorProps = {
+    style: Record<string, unknown> | Array<Record<string, unknown>>;
```

Comment:
I would consider more accurate type here:
```suggestion
    style: StyleProp<ViewStyle>
```